core/src/main/java/org/apache/iceberg/util/ThreadPools.java

```diff
@@ -86,9 +86,18 @@ public static ExecutorService newWorkerPool(String namePrefix) {
   }
 
   public static ExecutorService newWorkerPool(String namePrefix, int poolSize) {
-    return MoreExecutors.getExitingExecutorService(
-        (ThreadPoolExecutor)
-            Executors.newFixedThreadPool(poolSize, newDaemonThreadFactory(namePrefix)));
+    return newWorkerPool(namePrefix, poolSize, true);
+  }
+
+  public static ExecutorService newWorkerPool(
```

Comment:
Rather than introduce a boolean flag, which is hard to interpret.  I would suggest switch to use using two separate methods: `newWorkerPool`, `newExitingWorkerPool` (or something to that effect).  

The `newExitingWorkerPool` can just wrap the `newWorkerPool` method and return the exiting version.

Alternatively, we could invert this and add a `newNonExitingWorkerPool` since we want the default behavior to be exiting.

Reply:
Thanks @danielcweeks for the review!
@fengjiajie started a thread on the dev list about this topic: https://lists.apache.org/thread/mowmbr36y8wr1k9don2xx36l97n5f1xz

Cleaning up the pool in non-static cases should be a responsibility of the user. If they want a pool which is closed by a hook when the JVM exists they should explicitly "say" so, for example calling `newExitingWorkerPool`.

This is a behaviour change in the API, so I think we need feedback from the community before proceeding with it.
That said, we did not receive any answer for the thread. What do you think we should do to move forward?

Thanks,
Peter

Reply:
I agree with Dan's suggestion on adding a `newNonExitingWorkerPool` method that leaves the shutdown of the thread pool to users. Change some callers to this new API.

We don't change the behavior of current `newWorkerPool` which would continue to register shutdown hook.

We add clear Javadoc to explain the difference.